File: 2020-11/2020-11-23/coma.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 * Time Complexity 에서 꼴찌를 달성
 */
class Solution {
    public int rob(TreeNode root) {
        return Math.max(eat(root), noEat(root));
    }
    
    private int eat(TreeNode node) {
        if (node == null) return 0;
        int result = node.val;
        return result + noEat(node.right) + noEat(node.left);
    }
    
    private int noEat(TreeNode node) {
        if (node == null) return 0;
        int noLeft = noEat(node.left);
        int noRight = noEat(node.right);
        int yesLeft = eat(node.left);
        int yesRight = eat(node.right);
        int x = noLeft + noRight;
        int y = noLeft + yesRight;
        int w = yesLeft + noRight;
        int z = yesLeft + yesRight;
        return Math.max(Math.max(Math.max(x, y), z), w);
    }
}

File: 2020-12/2020-12-08/coma.py
class Solution:
    def numPairsDivisibleBy60(self, time: List[int]) -> int:
        d = [0] * 60
        for i in time:
            d[i % 60] += 1
        result = 0
        result += d[0] * (d[0] - 1) // 2
        result += d[30] * (d[30] - 1) // 2
        for i in range(1, 30):
            result += d[i] * d[60 - i]
        return result
